### DISCUSSION TITLE
graphql create data model with file inside?

### DISCUSSION THREAD

**Discussion by dunods** [directus/directus#11017] [Archive]
hello everyone 

need help regarding how to create data model with the file inside?

right now im having data model called user_document consist with user_id: integer and document:files type.

with current directus system, for create graphql for file system the code consist as below 

```graphql
mutation ($id: ID!) {
  create_user_documents_item(
    data: {user_id: {id: $id}, document: {storage: "", filename_download: "", uploaded_on: "", modified_on: ""}}
  ) {
    id
  }
}
```

but when i upload with the multipart method given from upload response and put it in above mutation no file was included

so how i make it happen by 
1. create_user_document_item by just putting user_id
2. upload the docs and get file_id

and last update update_user_document with response from step number 1 and response file id from number 2

is there any other simple way ??
please help

**Comment by rijkvanzanten** [comment]
> but when i upload with the multipart method given from upload response and put it in above mutation no file was included

How are you doing this exactly? If you set the `id` of the document, it should work as expected:

```graphql
mutation ($id: ID!) {
  create_user_documents_item(document: { id: "34e7c2e9-77a2-4ac6-a3d0-5beca3b2df41" }) {
    id
  }
}
```
